chains/manager.go

```diff
@@ -754,9 +755,7 @@ func (m *manager) createAvalancheChain(
 
 	chainAlias := m.PrimaryAliasOrDefault(ctx.ChainID)
 
-	untracedVMWrappedInsideProposerVM := &linearizeOnInitializeVM{
-		LinearizableVMWithEngine: vm,
-	}
+	untracedVMWrappedInsideProposerVM := NewLinearizeOnInitializeVM(vm)
```

Comment:
note: this does not use `dagVM`

Reply:
Is there a reason we didn't just add this as a comment in the code?

Reply:
(and to confirm, this is the only place we use `vm`)?

Reply:
I think adding it as a comment makes sense - this is the only place we use vm.

Reply:
```suggestion
	// Note: this does not use [dagVM] to ensure we use the [vm]'s height index.
	untracedVMWrappedInsideProposerVM := NewLinearizeOnInitializeVM(vm)
```